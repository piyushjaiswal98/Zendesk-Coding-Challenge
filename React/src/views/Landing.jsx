import React, { useEffect, useState } from 'react';
import './Landing.scss';
import axios from 'axios';
import Pageview from '../components/PageView';

const Landing = () => {

    const [all, setAll] = useState('active');
    const [individual, setIndividual] = useState('inactive');
    const [submit_button, setSB] = useState('inactive');
    const [data, setData] = useState([]);


    useEffect(() => {
        getTicketList();
    },[]);

    

    const getTicketList = () => {
        axios.get(global.config.uri + 'api/fetch')
            .then(res => {
                setData(res.data.tickets);
            })
            .catch(err => {
                console.log("Error: " + err);
            });
    }

    function handleAll(e) {
        e.preventDefault();
        setIndividual('inactive');
        setAll('active');
        setSB('inactive')
        document.getElementById('input-field').disabled = true;
    }

    function handleIndividual(e) {
        e.preventDefault();
        setAll('inactive');
        setIndividual('active');
        setSB('active');
        document.getElementById('input-field').disabled = false;
    }

    return(

        <div className="landing">

            <div className="landing-header">
                <p>Ticket Viewer</p>
            </div>

            <div className="container">
                <div className="sub-container">
                    <div className="box1">
                        <div className="box1-1">
                            <div className="lensToggle">
                                <div className={`b ${all}`} onClick={handleAll}><p>All Tickets</p></div>
                                <div className={`b ${individual}`} onClick={handleIndividual}><p>Filter</p></div>
                            </div>
                         </div>
                         <div className="box1-2">
                            <input type="text" id="input-field" placeholder="Input Ticket Id" disabled></input>
                            <button className={submit_button} id="submit_button" disabled>Submit</button>
                         </div>
                    </div>
                    <div className="box2">
                        <Pageview data={data}/>
                    </div>
                </div>
            </div>

        </div>
    )

}

export default Landing;

