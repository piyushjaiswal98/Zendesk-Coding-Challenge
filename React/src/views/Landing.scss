
.landing{
    width: 100%;
    min-height: 100vh;
    background-color: #282c34;

    .landing-header{
        width: 100%;
        height: 12vh;
        p{
            margin: 0;
            padding: 13px 0px 0px 0px;
            font-size: 2.3em;
            font-weight: bold;
            font-family: arial;
        }
        border-bottom: 1px solid white;
    }

    .container{
        width: 100%;
        min-height: 80vh;
        padding-top: 5vh;
        .sub-container{
            margin: 0% 20% 0% 20%;
            width: 60%;
            min-height: 72vh;
            // background-color: #282c34;
            // border: 1px solid white;
            .box1{
                width: 100%;
                height: 12vh;
                border-bottom: 1px solid #6c7483;
                display: flex;
                flex-direction: row;
                .box1-1{
                    flex: 0 0 50%;
                    .lensToggle{
                        // margin-right: 10px;
                        // margin-top: 2%;
                        // text-align: right;
                        // float: center;
                        margin: 5% 15% 5% 15%;
                        width: 250px;
                        background-color: rgb(211, 211, 211);
                        border: 1px solid #0099d6;
                        border-radius: 25px;
                        height: 6vh;
                        display: flex;
                        flex-wrap: wrap;
                        .b{
                            flex: 0 0 50%;
                            border-radius: 25px;
                            text-align: center;
                            font-size: 0.9em;
                            cursor: pointer;
                            color: rgb(50, 50, 50);
                            height: inherit;
                            // margin-top: 5%;
                            // padding-top: 5px;
                        }
                        p{
                            margin: 0 auto;
                            margin-top: 3px;
                        }
                        .inactive{
                            margin-top: 2px;
                        }                    
                        .active{
                            background-color: #0099d6;
                            // border: 1px solid #0099d6;
                            color: white;
                            font-weight: bold;
                            font-size: 0.9em;
                            height: 6vh ;
                            margin-top: -1px;
                        }
                    }
                    // border-right: 1px solid #6c7483;
                }
                .box1-2{
                    flex: 0 0 50%;
                    input{
                        width: 60%;
                        height: 5vh;
                        margin-top: 3vh;
                        outline: none;
                        border-radius: 5px;
                        border: 1px solid #0099d6;
                        margin-right: 5%;
                        font-size: 0.8em;
                        text-align: center;
                        // margin-left: -5%;
                    }
                    button{
                        outline: none;
                        border: none;
                        height: 5vh;
                        margin-top: 3.5vh;
                        border: 1px solid #0099d6;
                        font-size: 0.8em;
                        width: 25%;
                        border-radius: 5px;
                    }
                    button.active{
                        background-color: #0099d6;
                        color: white;
                        cursor: pointer;
                    }
                    button.inactive{
                        background-color: rgb(118, 118, 118);
                        color: rgb(80, 80, 80);
                        cursor: default;
                    }
                }
            }
            .box2{
                width: 100%;
                height: 60vh;
            }
        }
    }
}
//#6c7483