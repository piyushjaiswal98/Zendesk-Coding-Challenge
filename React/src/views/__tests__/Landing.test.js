import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Landing from '../Landing';

global.config = {
    uri: process.env.REACT_APP_API_HOST || 'http://localhost:2000/'
  }

afterEach(()=> {
    cleanup();
})

test('Should render Landing View', () => {
    render(<Landing/>);
    const LandingElement1 = screen.getByTestId('Landing-1');
    expect(LandingElement1).toBeInTheDocument();
    const LandingElement2 = screen.getByTestId('Landing-2');
    expect(LandingElement2).toBeInTheDocument();
    const LandingElement3 = screen.getByTestId('Landing-3');
    expect(LandingElement3).toBeInTheDocument();
})

test('Matches Snapshot', () => {
    const tree = renderer.create(<Landing/>).toJSON();
    expect(tree).toMatchSnapshot();
})