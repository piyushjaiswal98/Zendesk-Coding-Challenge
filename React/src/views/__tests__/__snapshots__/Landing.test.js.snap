// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div
  className="landing"
  data-testid="Landing-1"
>
  <div
    className="landing-header"
    data-testid="Landing-2"
  >
    <p>
      Ticket Viewer
    </p>
  </div>
  <div
    className="container"
    data-testid="Landing-3"
  >
    <div
      className="sub-container"
    >
      <div
        className="box1"
      >
        <div
          className="box1-1"
        >
          <div
            className="lensToggle"
          >
            <div
              className="b active"
              onClick={[Function]}
            >
              <p>
                All Tickets
              </p>
            </div>
            <div
              className="b inactive"
              onClick={[Function]}
            >
              <p>
                Filter
              </p>
            </div>
          </div>
        </div>
        <div
          className="box1-2"
        >
          <input
            disabled={true}
            id="input-field"
            onChange={[Function]}
            placeholder="Input Ticket Id"
            type="text"
            value=""
          />
          <button
            className="inactive"
            id="submit_button"
            onClick={[Function]}
          >
            Submit
          </button>
        </div>
      </div>
      <div
        className="box2"
      />
    </div>
  </div>
</div>
`;
